ifrs17/Import/4ImportScope-TechnicalMargin.ipynb

```diff
@@ -612,11 +635,61 @@
+                "\n                                             .WithApplicability<AllocateTechnicalMarginForReinsuranceCL>(x => x.Identity.AocType == AocTypes.CL)",
+                "\n                                             .WithApplicability<AllocateTechnicalMarginForReinsuranceEop>(x => x.Identity.AocType == AocTypes.EOP)",
+                "\n                                             .WithApplicability<AllocationForReinsurance>(x => true, //from here the applicabilities of the AllocateTechnicalMargin are applied with the following members redefined here.",
+                "\n                                                                                               p => p.ForMember(s => s.ComputedEstimateType)",
+                "\n                                                                                                     .ForMember(s => s.HasSwitch))",
+                "\n                                            );",
+                "\n    protected string ComputeReinsuranceEstimateType(double aggregatedFcf) => aggregatedFcf > Precision ? EstimateTypes.LR : EstimateTypes.C;",
+                "\n    protected double loReCoBoundaryValue => GetScope<LoReCoBoundary>(Identity).Value;",
+                "\n    protected double loReCoBoundaryAggregatedValue => GetScope<LoReCoBoundary>(Identity).AggregatedValue;",
+                "\n}",
+                "\n",
+                "\npublic interface AllocationForReinsurance : AllocateTechnicalMarginForReinsurance{",
+                "\n    [NotVisible] string AllocateTechnicalMargin.ComputedEstimateType => ComputeReinsuranceEstimateType(loReCoBoundaryAggregatedValue + loReCoBoundaryValue);",
+                "\n    [NotVisible] bool AllocateTechnicalMargin.HasSwitch => ComputedEstimateType != ComputeReinsuranceEstimateType(loReCoBoundaryAggregatedValue);",
+                "\n}",
+                "\n",
+                "\npublic interface AllocateTechnicalMarginForReinsurancePaa : AllocateTechnicalMarginForReinsurance{",
+                "\n    double AllocateTechnicalMargin.Value => EstimateType switch {",
+                "\n        EstimateTypes.LR => -1d * GetScope<LoReCoBoundary>(Identity).Value,",
+                "\n        _ => default",
+                "\n    };",
+                "\n}",
+                "\n",
+                "\npublic interface AllocateTechnicalMarginForReinsuranceCL : AllocateTechnicalMarginForReinsurance{   ",
+                "\n    string AllocateTechnicalMargin.ComputedEstimateType => ComputeReinsuranceEstimateType(loReCoBoundaryAggregatedValue + loReCoBoundaryValue);",
+                "\n    private double balancingValue => GetScope<PreviousAocSteps>((Identity, InputSource.Cashflow))",
+                "\n                                        .Values",
+                "\n                                        .GroupBy(x => x.Novelty)",
+                "\n                                        .Select(g => g.Last())",
```

Comment:
This means the last AOC Step for each novelty? Just checking...

Reply:
I would write the Select within the Group by, using the overload. I find it clearer
.GroupBy(x => x.Novelty, (k, v) => v.Last()) should work.

Are you sure this is ordered, right?